taf/src/main/java/com/taf/automation/locking/SystemDateManager.java

```diff
@@ -0,0 +1,624 @@
+    }
+
+    /**
+     * @param reservationInfo - Reservation Information to check against
+     * @return true if the Reservation Information is the least number of days to be added else false
+     */
+    private boolean isLeastAddDays(ReservationInfo reservationInfo) {
+        reservation.readLock().lock();
+        try {
+            if (reservationInfo.getState() == ReservationState.CLOCK_MOVED) {
+                // The ticket to move the clock has already been completed as such the clock will not be moved again
+                return true;
+            }
+
+            for (Map.Entry<Long, ReservationInfo> entry : storedReservations.entrySet()) {
+                ReservationState state = entry.getValue().getState();
+                Integer addDays = entry.getValue().getPlusDays();
+                if ((state == ReservationState.WAITING || state == ReservationState.RESERVED)
+                        && addDays != null && addDays < reservationInfo.getPlusDays()
+                ) {
+                    return false;
+                }
+            }
+
+            return true;
+        } catch (Exception ex) {
+            return false;
+        } finally {
+            reservation.readLock().unlock();
+        }
+    }
+
+    /**
+     * Update the existing reservations to account for the processed reservation
+     *
+     * @param reservationInfo - Reservation that was processed
+     */
+    private void updateReservations(ReservationInfo reservationInfo) {
+        reservation.writeLock().lock();
+        try {
+            for (Map.Entry<Long, ReservationInfo> entry : storedReservations.entrySet()) {
+                if (reservationInfo.getPlusDays().equals(entry.getValue().getPlusDays())) {
+                    entry.getValue().setState(ReservationState.CLOCK_MOVED);
+                } else if (entry.getValue().getState() == ReservationState.WAITING ||
+                        entry.getValue().getState() == ReservationState.RESERVED
+                ) {
+                    // Note:  We are only moving the clock on the smallest number of days as such this is always positive
+                    entry.getValue().setPlusDays(entry.getValue().getPlusDays() - reservationInfo.getPlusDays());
+                }
+            }
+        } finally {
+            reservation.writeLock().unlock();
+        }
+    }
+
+    /**
+     * Move the clock for this reservation
+     *
+     * @param ticket          - Ticket that the clock is being moved for
+     * @param reservationInfo - Reservation to be move the clock
+     */
+    private void moveTheClock(Long ticket, ReservationInfo reservationInfo) {
+        action.lock();
+        try {
+            moveClockAction.accept(ticket, reservationInfo.getPlusDays());
+        } finally {
+            action.unlock();
+        }
+    }
+
+    /**
+     * Close the reservation ticket
+     *
+     * @param ticket - Ticket to be closed
+     */
+    public void closeReservation(Long ticket) {
```

Comment:
For debugging purposes, this method should return true if the ticket was closed or false otherwise (invalid ticket or perhaps already closed.)